GitUI/CommandsDialogs/BuildReportTabPageExtension.cs

```diff
@@ -111,6 +95,58 @@ public void FillBuildReport([CanBeNull] GitRevision revision)
             }
         }
 
+        private void LoadReportContent(GitRevision revision, bool isFavIconMissing)
+        {
+            try
+            {
+                if (revision.BuildStatus.ShowInBuildReportTab)
+                {
+                    _buildReportWebBrowser.Navigate(revision.BuildStatus.Url);
+                }
+
+                if (isFavIconMissing)
+                {
+                    _buildReportWebBrowser.Navigated += BuildReportWebBrowserOnNavigated;
+                }
+            }
+            catch
+            {
+                // No propagation to the user if the report fails
+            }
+        }
+
+        private void SetTabPageContent(GitRevision revision)
+        {
+            if (revision.BuildStatus.ShowInBuildReportTab)
+            {
+                _url = null;
+                Control = _buildReportWebBrowser;
+                _buildReportTabPage.Controls.Add(_buildReportWebBrowser);
+            }
+            else
+            {
+                _url = revision.BuildStatus.Url;
+                _buildReportTabPage.Cursor = Cursors.Hand;
+                Control = _openReportLink;
+                _buildReportTabPage.Controls.Add(_openReportLink);
+            }
```

Comment:
The same issue here - we will be adding controls over and over again.
The controls must be added once, and then we should just toggle their visibility.

Reply:
The controls are generated only once and `_buildReportTabPage.Controls` is cleared, so here also, I don't think we leak memory.

>The controls must be added once, and then we should just toggle their visibility.

I think that won't work due to how controls are placed (dock: fill):

https://github.com/gitextensions/gitextensions/pull/7224/files#diff-1dc38168f8d6f6714647e795cc45f511R28

I made this to add this features: "link centered" && "open report wherever the user click on the tab page"

Reply:
Thank you